crates/subspace-farmer/src/archiving.rs

```diff
@@ -96,6 +98,7 @@ impl Archiving {
         let (new_block_to_archive_sender, new_block_to_archive_receiver) =
             std::sync::mpsc::sync_channel::<Arc<AtomicU32>>(0);
         let (stop_sender, stop_receiver) = oneshot::channel::<()>();
+        let (archived_blocks_sender, _) = broadcast::channel(1);
```

Comment:
This doesn't seem to be a good idea (both broadcast and capacity of 1). It will simply drop older messages if for some reason receivers are not able to keep up with producer. The fact that `send` is not async made me question what capacity does so I went to read docs.

Something like `mpmc` channel (or analog) is necessary where receiver can be returned and cloned later, such that we can start as many consumers as needed before dropping original consumer and let the usual logic of `send` returning error on last consumer dropped kick in at that point.

In this case you could have not dropped receiver and carried it over so that `send` doesn't fail below, but it still wouldn't solve capacity behavior of the broadcast channel.